velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -210,10 +211,17 @@ namespace filesystems {
 
 class S3WriteFile::Impl {
  public:
-  explicit Impl(const std::string& path, Aws::S3::S3Client* client)
+  explicit Impl(
+      const std::string& path,
+      Aws::S3::S3Client* client,
+      memory::MemoryPool* pool)
       : client_(client) {
+    VELOX_CHECK_NOT_NULL(client);
+    VELOX_CHECK_NOT_NULL(pool);
     getBucketAndKeyFromS3Path(path, bucket_, key_);
-
+    pool_ = pool->addLeafChild(".s3writefile");
```

Comment:
```
// TODO: we need to set memory reclaimer of the created leaf memory pool to integrate with memory arbitrator later.
```

Reply:
No need for this since we are passing a leaf pool already here. I removed the leaf pool creation.